apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_impl.erl

```diff
@@ -0,0 +1,355 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+-module(emqx_bridge_iotdb_impl).
+
+-include("emqx_bridge_iotdb.hrl").
+-include_lib("emqx/include/logger.hrl").
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
+
+%% `emqx_resource' API
+-export([
+    callback_mode/0,
+    on_start/2,
+    on_stop/2,
+    on_get_status/2,
+    on_query/3,
+    on_query_async/4
+]).
+
+-type config() :: map().
+
+%% [FIXME] should be emqx_connetor_http:state() or wrapped, but instead for now
+%% copy-pasta.
+-type state() ::
+    #{
+        pool_name := any(),
+        %% [optional] pool_type := round_robin | random | hash | direct | claim
```

Comment:
Currently, `emqx_connector_http` only supports `hash` and `random`.

Reply:
I now document only what's needed by the module itself, it's mostly meant as a helper to the developer to know what needs to be there and of what type it is. Don't see much value documenting what `emqx_connector_http` expects. This was more of a relic of a very early stage of my development cycle. 😅